concrete/dense_directed_matrix.go

```diff
@@ -0,0 +1,135 @@
+// Copyright ©2014 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package concrete
+
+import (
+	"github.com/gonum/graph"
+	"github.com/gonum/matrix/mat64"
+)
+
+// DirectedDenseGraph represents a graph such that all IDs are in a contiguous
+// block from 0 to n-1.
+type DirectedDenseGraph struct {
+	absent float64
+	mat    *mat64.Dense
+}
+
+// NewDirectedDenseGraph creates a directed dense graph with n nodes.
+// If passable is true all nodes will have an edge with unit cost, otherwise
```

Comment:
"all nodes will have an edge" -> "all pairs of nodes will be connected by an edge" or something similar?